### ISSUE TITLE
SaveDialog Qt5 connect warning and error on cancel

### ISSUE THREAD

**Issue by pllim** [ejeschke/ginga#476]
Qt5 on RHEL7.

1. Open an image.
2. Open Cuts plugin.
3. Draw a cut to enable Save button.
4. Click on Save button.

When button is clicked, I see this on the terminal:
```
QObject::connect: invalid null parameter
```

Then I click Cancel without saving the file and I see this:
```python
Error making callback 'activated': can only concatenate tuple (not "str") to tuple
Traceback:
  File ".../ginga/misc/Callback.py", line 123, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga/rv/plugins/Cuts.py", line 210, in <lambda>
    lambda w: self.save_cb(mode='cuts'))

  File ".../ginga/rv/plugins/Cuts.py", line 1015, in save_cb
    self.fv.error_wrap(image.save_as_file, target + '.fits')
```

The problem with the second traceback is that this is no longer a valid check:
```python
        # Save cancelled
        if not target:
            return
```

**Comment by ejeschke** [comment]
I think I recently had to add a fix to check whether `target` is an instance of `tuple`, and if so, the filename is element 0 of said tuple.  Sounds like we need to refactor this code a little to check exactly what is coming in as `target`.

**Comment by ejeschke** [comment]
This is partially fixed by commit 2e5846c98c89074bbd8c2cd8a4fd4cfa9852d094.

There is still an error that appears on the terminal when canceling:
```python
setNativeLocks failed: Resource temporarily unavailable
```
I don't know if this is a platform-specific error, but it doesn't seem to cause any particular problem.

Also, in this commit I removed a save of the entire image data to a FITS file.  I didn't quite see the point in this since the data is already in the image file.  If the user really needs to make a copy they can use the `SaveImage` plugin, I think.

**Comment by pllim** [comment]
@ejeschke , I don't see the `setNativeLocks failed` message on my end and can confirm that your commit removed the original reported traceback. I'll see if I can find more of such cases. Thanks!
